src/snarky/parties-leaves.ts

```diff
@@ -0,0 +1,249 @@
+import { Field, Bool, Group, Ledger } from '../snarky';
```

Comment:
what is the difference between this file and `parties-leaves-json`? seems like they can share more content -- are these autogenerated?

Reply:
No these are written manually!

There are two "Party" types - one which directly corresponds to the JSON schema, which is (autogenerated) in `parties-json.ts`, and one which corresponds to our existing snarkyjs types -- so it can be used in a circuit -- which is (autogenerated) in `parties.ts`.
For both of those types, we need to manually specify what the primitive types, i.e. the "scalars" in the graphql schema, should be. This is what's done in the `-leaves` files. So, for example,
* `parties-leaves-json.ts` declares that "Fee" should become `string`
* `parties-leaves.ts` declares that "Fee" should become `UInt64`

In addition, `parties-leaves.ts` has manually written methods to convert from the snarkyjs leaf types to the JSON leaf types, e.g. it says that the method to convert `UInt64` to `string` is `x => x.toString()`.

Similarly, `parties-leaves.ts` has manually written converters of leaf types to `Field[]`, e.g. `(x: UInt64) => x.toFields()`.

Reply:
and to answer your question: the two files have no potential for code sharing that is worth doing IMO

Reply:
can you put this response into a comment on one of the files? (and on the other one say something like "see parties-leaves.ts for more details on what this is for")